telekom-workflow-engine/src/main/java/ee/telekom/workflow/executor/lifecycle/LifecycleServiceImpl.java

```diff
@@ -166,6 +169,9 @@ else if( NodeStatus.FAILED.equals( node.getStatus() ) ){
                 // happen to get to know this, we will shutdown
                 stopIfRunning();
             }
+
+            // Step 4: Clean up archive entries
+            archiveService.cleanup();
```

Comment:
Forgot to discuss earlier, but we need to make it so tha only the "cluster master node" will run archive cleanup. Otherwise multiple nodes could run it at the same time and select the same ref_num-s and try to delete the same rows - inefficient.

So we should call a private method, similar to Step 2 healFailedNodesIfLockedOwned() here. Inside it do an 
if( lockService.eagerAcquire() ){
    // this is the master node, run the archive cleanup
    archiveService.cleanup();
}